cmd/podman/containers/cp.go

```diff
@@ -39,25 +54,312 @@ var (
 
 func cpFlags(cmd *cobra.Command) {
 	flags := cmd.Flags()
-	flags.BoolVar(&cpOpts.Extract, "extract", false, "Extract the tar file into the destination directory.")
-	flags.BoolVar(&cpOpts.Pause, "pause", true, "Pause the container while copying")
+	flags.BoolVar(&cpOpts.Extract, "extract", false, "Deprecated...")
+	flags.BoolVar(&cpOpts.Pause, "pause", true, "Deorecated")
+	_ = flags.MarkHidden("extract")
+	_ = flags.MarkHidden("pause")
 }
 
 func init() {
 	registry.Commands = append(registry.Commands, registry.CliCommand{
-		Mode:    []entities.EngineMode{entities.ABIMode},
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
 		Command: cpCommand,
 	})
 	cpFlags(cpCommand)
 
 	registry.Commands = append(registry.Commands, registry.CliCommand{
-		Mode:    []entities.EngineMode{entities.ABIMode},
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
 		Command: containerCpCommand,
 		Parent:  containerCmd,
 	})
 	cpFlags(containerCpCommand)
 }
 
 func cp(cmd *cobra.Command, args []string) error {
-	return registry.ContainerEngine().ContainerCp(registry.GetContext(), args[0], args[1], cpOpts)
+	// Parse user input.
+	sourceContainerStr, sourcePath, destContainerStr, destPath, err := copy.ParseSourceAndDestination(args[0], args[1])
+	if err != nil {
+		return err
+	}
+
+	if len(sourceContainerStr) > 0 {
+		return copyFromContainer(sourceContainerStr, sourcePath, destPath)
+	}
+
+	return copyToContainer(destContainerStr, destPath, sourcePath)
+}
+
+// containerMustExist returns an error if the specified container does not
+// exist.
+func containerMustExist(container string) error {
+	exists, err := registry.ContainerEngine().ContainerExists(registry.GetContext(), container, entities.ContainerExistsOptions{})
+	if err != nil {
+		return err
+	}
+	if !exists.Value {
+		return errors.Errorf("container %q does not exist", container)
+	}
+	return nil
+}
```

Comment:
Maybe I have the exists() function from the new API stuck in my head, but I thought we'd similar functionality elsewhere in libpod?  I looked but didn't find it quickly.  This feels like a function that should live in util.

This is fine here for now, but something to ruminate on.

Reply:
Yes, there's a ContainerExists function.  I thought to put that into  a dedicated function (plus the checks) to consolidate the logic a bit further.